xla/service/latency_hiding_scheduler_test.cc

```diff
@@ -3074,6 +3088,51 @@ ENTRY AddR2 {
   XLA_VLOG_LINES(1, hlo_module->ToString());
 }
 
+// Function to retrieve the value for a specific key from HloPassMetadata
+// for a given pass_name. Returns 0 if the key or pass_name is not found.
+int64_t GetKeyValueFromHloPassMetadata(const HloModuleMetadataProto& metadata,
+                                       const std::string& pass_name,
+                                       const std::string& target_key) {
+  for (const auto& pass_metadata : metadata.pass_metadata()) {
+    if (pass_metadata.pass_name() == pass_name) {
+      for (const auto& kv_metric : pass_metadata.kv_metrics()) {
+        if (kv_metric.key() == target_key) {
+          return kv_metric.value();
+        }
+      }
+    }
+  }
+  return 0;
+}
+
+TEST_F(LatencyHidingSchedulerTest, GetGPUPeakMemoryUsage) {
+  // Retrieve GPU peak memory usage and verify it excludes host memory
+  absl::string_view kPeakMemoryUsageWithCpuOffload = R"(
```

Comment:
For consistency with the other tests:
```suggestion
  absl::string_view hlo_string = R"(
```